From b5eb1c24e8399468390178d8ad3103b941cfe186 Mon Sep 17 00:00:00 2001
From: CI Assistant <chdauto@intel.com>
Date: Thu, 6 Feb 2020 14:34:52 +0200
Subject: [PATCH] Merge pull request #1737 in SW_UGW/linux from
 bugfix/UGW_SW-47271-ppa-acceleration-fix-8.4.1 to 8.4.1

* commit '116425a267796ee14b1e6b17535b45063f0890b4':
  UGW_SW-47271 : PPA acceleration fix
---
 include/linux/skbuff.h | 6 +++---
 net/core/skbuff.c      | 3 ---
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index f77a08e1ec11..af9df71994c5 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -809,6 +809,9 @@ struct sk_buff {
 #if defined(CONFIG_LTQ_CBM) || IS_ENABLED(CONFIG_DIRECTCONNECT_DP_API)
 	__u32 DW0, DW1, DW2, DW3;
 #endif
+#ifdef CONFIG_NETWORK_EXTMARK
+	__u32		extmark;
+#endif
 #ifdef CONFIG_PPA
 	void			*ptr_ppa_pitem;
 #endif
@@ -824,9 +827,6 @@ struct sk_buff {
 				*data;
 	unsigned int		truesize;
 	atomic_t		users;
-#ifdef CONFIG_NETWORK_EXTMARK
-	__u32		extmark;
-#endif
 };
 
 #ifdef __KERNEL__
diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index 037bb4a4af69..07a1ba971aa0 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -886,9 +886,6 @@ static void __copy_skb_header(struct sk_buff *new, const struct sk_buff *old)
 	memcpy(&new->headers_start, &old->headers_start,
 	       offsetof(struct sk_buff, headers_end) -
 	       offsetof(struct sk_buff, headers_start));
-#ifdef CONFIG_NETWORK_EXTMARK
-	new->extmark	 = old->extmark;
-#endif
 #if IS_ENABLED(CONFIG_PPA)
 	if (ppa_hook_pitem_refcnt_inc_fn && new->ptr_ppa_pitem)
 		ppa_hook_pitem_refcnt_inc_fn(new);
